trpcultivate_germcollection/tests/src/Kernel/TripalImporter/GermplamRelationshipImporter/GermplasmRelationshipImporterFormTest.php

```diff
@@ -0,0 +1,277 @@
+    // We expect there to be an upload description includinf a template link
+    // and numbered column description.
+    $this->assertArrayHasKey('upload_description', $form['file'], 'We expect the upload description to be added by TripalImporte base class.');
+    $this->assertStringContainsString('<a href', $form['file']['upload_description']['#markup'], "We expected the upload description to have a link in it.");
+    $this->assertStringContainsString('<ol id="tcp-header-notes">', $form['file']['upload_description']['#markup'], "We expected the upload description to have an ordered list in it.");
+    // We also expect the file upload HTML5 element provided by Tripal
+    // and not the file local/remote.
+    $this->assertArrayHasKey('file_upload', $form['file'],
+      "We expect the file upload element to be added by the Tripal Importer base class.");
+    $this->assertArrayNotHasKey('file_local', $form['file'],
+      "The local file element should not be available.");
+    $this->assertArrayNotHasKey('file_remote', $form['file'],
+      "The remote file element should not be available.");
+
+    // Check the population entry field element.
+    $this->assertArrayHasKey('fieldset_population_entry', $form,
+      'We expect there to be a population entry form element on the form but there is not.');
+    $this->assertEquals('fieldset', $form['fieldset_population_entry']['#type'],
+      'We expect the population element for the title in the form to be a fieldset.');
+    $this->assertArrayHasKey('fld_text_population_entry', $form['fieldset_population_entry'], 'We expect there to be a textfield for entering the population entry.');
+    $this->assertEquals('textfield', $form['fieldset_population_entry']['fld_text_population_entry']['#type'],
+      'We expect the population entry element in the form to be a textfield.');
+
+    // Check the Relationship type element.
+    $this->assertArrayHasKey('fieldset_relationship_type', $form,
+      'We expect there to be a relationship type form element on the form but there is not.');
+    $this->assertEquals('fieldset', $form['fieldset_relationship_type']['#type'],
+      'We expect the relationship type element in the form to be a fieldset.');
+    $this->assertArrayHasKey('fld_select_relationship_verb', $form['fieldset_relationship_type'], 'We expect there to be an textfield for entering the relationship type.');
+    $this->assertEquals('textfield', $form['fieldset_relationship_type']['fld_select_relationship_verb']['#type'],
+      'We expect the relationship type element in the form to be a textfield.');
+    $this->assertArrayHasKey('fld_radio_stock_position', $form['fieldset_relationship_type'], 'We expect there to be a radio button for selecting the stock position.');
+    $this->assertEquals('radios', $form['fieldset_relationship_type']['fld_radio_stock_position']['#type'], 'We expect the stock position element in the form to be a set of radio buttons.');
+  }
```

Comment:
```suggestion
     
     // Check the Relationship toggle element.
    $this->assertArrayHasKey('relationship_toggle', $form, 'We expect there to be an checkbox for relationship toggle field.');
    $this->assertEquals('checkbox', $form['relationship_toggle']['#type'], 'We expect the relationship toggle element in the form to be a checkbox.');
    $this->assertEquals(0, $form['relationship_toggle']['#default_value'], 'We expect the relationship toggle element in the form to be set to false by default.');
  }
```
To test the toggle form element.